File: src/main/java/com/kiwiko/library/metrics/api/Logger.java
package com.kiwiko.library.metrics.api;

import com.kiwiko.library.metrics.data.Log;

public interface Logger {

    void debug(Log log);
    void debug(String message);
    void debug(String message, Throwable cause);

    void info(Log log);
    void info(String message);
    void info(String message, Throwable cause);

    void warn(Log log);
    void warn(String message);
    void warn(String message, Throwable cause);

    void error(Log log);
    void error(String message);
    void error(String message, Throwable cause);
}

File: react/src/tools/monitoring/logging/Logger.js


export default class Logger {

    log(level, message, error) {
        throw new Error('No implementation for log()');
    }

    error(message, error) {
        this._writeLog('error', message, error);
    }

    warn(message, error) {
        this._writeLog('warn', message, error);
    }

    info(message, error) {
        this._writeLog('info', message, error);
    }

    debug(message, error) {
        this._writeLog('debug', message, error);
    }

    _writeLog(level, message, error) {
        let messageText = message;
        if (typeof message === 'function') {
            messageText = message();
        }
        this.log(level, messageText, error);
    }
}